Fix slow wallet deletion and db issues

Anviking opened the pull request:
# Issue Number

ADP-827

# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- [x] Don't retry on SQLITE_BUSY when trying to close

# Comments

- The retrying would timeout anyway, leaving the connection open and the file un-deleted.

## Alternative attempts / notes
- `.deleted` lock and removing `waitForFree` lead to err500 on wallet creation.
- (On master) it seemed to consistently (100% of the time) get stuck in destroySqliteBackend trying to close, _when_ a db query was cancelled.

## Effects on tests

With `-j 8`
- 240s
- 0 >10s DELETE or POST /wallet requests

<!-- Additional comments or screenshots to attach if any -->

<!--
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Jira will detect and link to this PR once created, but you can also link this PR in the description of the corresponding ticket
 ✓ Acknowledge any changes required to the Wiki
 ✓ Finally, in the PR description delete any empty sections and all text commented in <!--, so that this text does not appear in merge commit messages.
-->

Anviking commented on lib/core/src/Cardano/DB/Sqlite.hs:
```diff
@@ -236,25 +238,6 @@ destroySqliteBackend
     -> IO ()
 destroySqliteBackend tr sqlBackend dbFile = do
```
Function is now misleading. Should remove if we go with this solution.

rvl reviewed (commented):
Wouldn't this prevent sqlite db connections from ever being closed, unless there is an error with a query? That would be a resource leak.

Anviking committed:
Don't retry to close sqlite connections

Hypothesis: the retrying has no effect, so we might as well fail fast.

Looking at the integration test wallet dir with NO_CLEANUP=1 it seems
the wallet with busy connections are left un-deleted both with this
commit and on master.
